dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunction.cs

```diff
@@ -74,7 +74,7 @@ public abstract class KernelFunction
     /// <summary>
     /// Gets the prompt execution settings.
     /// </summary>
-    internal List<PromptExecutionSettings>? ExecutionSettings { get; }
+    public List<PromptExecutionSettings>? ExecutionSettings { get; }
```

Comment:
IReadOnlyList? Otherwise this is mutable.

Reply:
Also, is it OK that the execution settings themselves are mutable via this?

Reply:
Yeah worth investigating to see can everything we made read only to start with

Reply:
Created this task to track https://github.com/microsoft/semantic-kernel/issues/4224